### PR TITLE
[docs] Improve Git workflow documentation

### LINKED ISSUE #80: Add more Git workflow documentation for contributors

### PR DESCRIPTION
**Pull request by mihaic**
Fixes #80.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -31,10 +31,50 @@ in mind:
 How to contribute
 =================
 
-We use GitHub pull requests (PRs) to make improvements to the repository.
+We use GitHub pull requests (PRs) to make improvements to the repository. Here
+is a concise description of our recommended workflow:
```
Comment (yidawang) [inline CONTRIBUTING.rst]: We may want to say: Here is a concise step-by-step description of our recommended workflow. It is worth noting that the workflow is not unique. In general, we suggest contributors creating a new branch in the downstream and work on it to merge to our `master` branch.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -31,10 +31,50 @@ in mind:
...
+
+     git pull --ff-only
+
+6. Always create a new branch when you start working on a new feature; we only
+   update the ``master`` branch via pull requests from feature branches; never
+   commit directly to the ``master`` branch::
+
+     git checkout -b new-feature
+
+7. Push your feature branch to your fork::
```
Comment (yidawang) [inline CONTRIBUTING.rst]: `git add` and `git commit` before `git push`?

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -31,10 +31,50 @@ in mind:
...
+     git push --set-upstream origin new-feature  # only for the first push
+     git push  # for all subsequent pushes
+
+8. When your feature is ready, make a pull request on GitHub. After your
+   feature is accepted, update your ``master`` branch and delete your feature
+   branch::
+
+     git checkout master
+     git pull --ff-only
+     git branch -d new-feature
```
Comment (yidawang) [inline CONTRIBUTING.rst]: This only deletes the local branch. `git push --delete origin new-feature` will delete the branch on github.

Comment (mihaic) [inline CONTRIBUTING.rst]: All done.

**Pull request merged**